test/utils.sh

```diff
@@ -823,23 +823,26 @@ replace_image_pullspec() {
   fi
 
   local image_regex="^([^:@]+)(:[^@]+)?(@sha256:[a-f0-9]{64})?$"
-  if [[ "$image" =~ $image_regex ]]; then
-    local digest=""
-    if [[ "$image" =~ (@sha256:[a-f0-9]{64}) ]]; then
-      digest=$(echo "$image" | sed -E 's/^.*(@sha256:[a-f0-9]{64})$/\1/')
-      image=${image%%@*}
-    fi
-
-    local tag=""
-    if [[ "$image" =~ (:[^@]+) ]]; then
-      tag=$(echo "$image" | sed -E 's/^.*(:[^@]+).*$/\1/')
-    fi
-
-    echo "${mirror}${tag}${digest}"
-  else
+  if ! [[ "$image" =~ $image_regex ]]; then
     echo "replace_image_pullspec: invalid pullspec format: ${image}" >&2
     exit 2
   fi
```

Comment:
Do we need this or can we just rely on the validation of `parse_image_url`? Do we need to improve the validation of that function?

Reply:
I tried relying on the validation in `parse_image_url` but one of the [tests](https://github.com/konflux-ci/konflux-test/blob/main/unittests_bash/test_utils.bats#L525)  that verifies the sha is of the required length fails. 

Enhancing the validation to that function causes nearly 20 other tests to fail mainly because dummy sha's are used in the unit tests.

I can fix the validation and tests in parse_image_url and update the PR

Reply:
Ah. Given the fact that we have encountered multiple bugs due to test cases not accurately reflecting the real world, while this is a non-ideal case, I think the change might be beneficial in the long run.

Reply:
I agree with @arewm, if the test cases don't reflect the real world, they should be updated.